### PR TITLE
refactor: move settings in app

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-524

### User facing changelog
- Move the "settings" page to the app
- Polish all the screens connected
- Wire it to all the available GraphQL data

### Additional details
- I also took the liberty to wire the record keys to the "empty runs" page

### How has the user experience changed?

Screen Shot 2021-11-02 at 5 49 45 PM
Screen Shot 2021-11-02 at 5 49 36 PM
Screen Shot 2021-11-02 at 5 49 27 PM
Screen Shot 2021-11-02 at 5 49 09 PM
Screen Shot 2021-11-02 at 5 49 02 PM

### To test it

- `yarn dev`
- open the "app" project in the cypress repo
- open component testing
- open `__vite__/#/`
- navigate to the settings with the left navigation bar.

### Caveats

- The device settings are not wired up to GQL since the gql values don't exist
- `specPatterns` don't exist yet so the value is there just for show
- `experiments` are not wired either

### CODE REVIEW

**Review by lmiller1990 — COMMENTED** [review]
Some comments!

It looks great, verified it:

image

File: packages/app/cypress/e2e/integration/settings.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    cy.initializeApp()
+  })
+
+  it('displays the settings for the current project', () => {
+    cy.visitApp()
+    cy.get('[href="#/settings"]').click()
+    cy.findByText('Project Settings').click()
+
+    cy.findByText('projectId')
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/app/cypress/e2e/integration/settings.spec.ts]: Fixed in commit f89aab5

File: packages/app/src/runs/RunsEmpty.vue [hunk 2]
```diff
@@ -77,6 +82,14 @@ const projectIdCode = computed(() => {
 
 const projectName = computed(() => props.gql.title)
 const configFilePath = computed(() => props.gql.configFilePath)
+const firstRecordKey = computed(() => {
+  const allRecordKeys = props.gql.cloudProject?.recordKeys
+
+  return allRecordKeys?.length ? allRecordKeys[0] : '<record-key>'
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/app/src/runs/RunsEmpty.vue:88]: Done in e7a1872d974616ef56fe69ffdd930a50de62bf67

File: packages/frontend-shared/src/components/ListRowHeader.vue [hunk 3]
```diff
@@ -1,39 +1,36 @@
...
+      <p class="text-sm font-light select-none text-gray-500">
+        <slot name="description">
+          <span>{{ description }}</span>
+        </slot>
+      </p>
+    </div>
     <div
-      class="flex items-center w-full text-left rounded py-12px"
-      :class="props.class"
+      v-if="$slots.right"
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/frontend-shared/src/components/ListRowHeader.vue:29]: Done in 35da10109dbfef7f4bab2dc9e19db0e058d2c1b0

File: packages/app/package.json [hunk 4]
```diff
@@ -72,30 +72,66 @@
...
         "@urql/vue",
         "@vueuse/core",
+        "arch",
+        "bluebird",
+        "cachedir",
+        "chalk",
+        "check-more-types",
+        "cli-cursor",
+        "cli-table3",
+        "commander",
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/app/package.json:90]: Probably a mistake at some point made all the node stuff go through vite. I removed it in d455ca929f5c65f562ac72d457f4acc47c0895b5

File: packages/app/src/settings/device/ExternalEditorSettings.vue [hunk 5]
```diff
@@ -85,5 +85,5 @@ const externalEditors = [
 ]
 
 const { t } = useI18n()
-const selectedEditor = ref()
+const selectedEditor = ref<Record<string, any>>()
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/app/src/settings/device/ExternalEditorSettings.vue:88]: I agree, but I figured these types should all come from GQL anyway so I left it as the simplest iteration.

File: packages/app/src/settings/SettingsCard.spec.tsx [hunk 6]
```diff
@@ -12,7 +12,7 @@ describe('<SettingsCard />', () => {
 
     cy.mount(() => (
       <div class="">
-        <SettingsCard title={title} description={description} icon={IconLaptop}>
+        <SettingsCard title={title} description={description} icon={IconLaptop} maxHeight="800px">
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/app/src/settings/SettingsCard.spec.tsx:15]: I really don't want the whole style to be modifiable here. This is only an attribute meant to enable the smooth expansion of the collapsible. 

What do you think?

File: packages/frontend-shared/src/components/ListRowHeader.vue [hunk 7]
```diff
@@ -1,39 +1,36 @@
 <template>
-  <div>
+  <div
+    class="flex items-center w-full text-left rounded py-12px"
+    :class="props.class"
```
[the remark this answers is not in the document]
Comment (elevatebart) [reply packages/frontend-shared/src/components/ListRowHeader.vue]: I don't even think we need the `inheritAttrs`.
This was an artifact of when the class was passed to a lower-level item.

Removed in d6dc8a7

**Comment by marktnoonan** [comment]
I added a couple of style tweaks to this branch, looks great when I run it, did not get to do a full code review.

File: packages/app/src/settings/SettingsCard.spec.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (lmiller1990) [reply packages/app/src/settings/SettingsCard.spec.tsx:15]: No strong opinion, happy to go with this if you think it makes sense!

**Pull request merged**